samples/net/openthread/coap_server/README.rst

```diff
@@ -1,13 +1,14 @@
 .. _coap_server_sample:
 
-openthread: coap_server
-##################
+OpenThread: CoAP Server
+#######################
 
-The openthread coap_server sample demonstrates interactions between nodes performing different Thread roles with the use of OpenThread and CoAP protocol.
+The OpenThread CoAP Server sample demonstrates interactions between nodes performing different Thread roles with the use of OpenThread and CoAP protocol.
 
 Overview
 ********
-
+You can use this sample application as a starting point to implement a CoAP application.
```

Comment:
OK, but please explain what a CoAP application is, even just by expanding the acronym.
And provide more technical details. This needs to be way more detailed than the introduction, if possible.